packages/fx-core/src/plugins/resource/frontend/ops/deploy.ts

```diff
@@ -51,60 +52,104 @@ export class FrontendDeployment {
     return lastDeployTime < lastBuildTime;
   }
 
-  public static async doFrontendBuild(
-    componentPath: string,
-    envs: RemoteEnvs,
-    envName: string
-  ): Promise<void> {
+  public static async doFrontendBuild(componentPath: string, envName: string): Promise<void> {
     if (!(await FrontendDeployment.needBuild(componentPath, envName))) {
       return FrontendDeployment.skipBuild();
     }
 
     const progressHandler = ProgressHelper.progressHandler;
 
+    const scripts =
+      (await runWithErrorCatchAndWrap(
+        (error) => new FileIOError(error.message),
+        async () => (await fs.readJSON(path.join(componentPath, PathInfo.NodePackageFile))).scripts
+      )) ?? [];
+
+    if (!("install:teamsfx" in scripts)) {
+      // * Track legacy projects
+      TelemetryHelper.sendGeneralEvent(TelemetryEvent.InstallScriptNotFound);
+    }
+
     await progressHandler?.next(DeployProgress.steps.NPMInstall);
     await runWithErrorCatchAndThrow(new NpmInstallError(), async () => {
-      await Utils.execute(Commands.InstallNodePackages, componentPath);
+      await Utils.execute(
+        "install:teamsfx" in scripts
```

Comment:
How about the following?
`
    execute(scripts["install:teamsfx"] ?? DefaultScript)
`

Reply:
A little different. 
```
await Utils.execute(
        "install:teamsfx" in scripts
          ? Commands.InstallNodePackages
          : Commands.DefaultInstallNodePackages,
        componentPath
      );
```
This output logs: `Execute command: npm run install:teamsfx` so that user knows how to customize install script by reading logs.
```
execute(scripts["install:teamsfx"] ?? DefaultScript)
```
This output: `Execute command: npm install` by default. User may not get how to customize it.